File: tests/code_index/fixtures/java/Extras.java
package demo;

/** Extras for DESIGN §12.1 coverage — generic class, variadic, abstract. */
public class Extras<T> {
    public T identity(T x) { return x; }

    public int sumAll(int first, int... rest) {
        int s = first;
        for (int v : rest) s += v;
        return s;
    }
}

abstract class AbstractBase {
    public abstract int absMethod();
}

File: tests/code_index/fixtures/go/extras.go
// Extras for DESIGN §12.1 coverage — generic function (Go 1.18+),
// variadic, type reference checks.
package demo

// Variadic — last param has `...`.
func Variadic(prefix string, vals ...int) int {
	total := 0
	for _, v := range vals {
		total += v
	}
	return total
}

// Generic function with type parameter.
func Identity[T any](x T) T {
	return x
}
